include/mxnet/tensor_blob.h

```diff
@@ -0,0 +1,773 @@
+/*!
+ *  Copyright (c) 2014 by Contributors
+ * \file tensor_blob.h
+ * \brief TBlob class that holds common representation of
+ *  arbirary dimension tensor, can be used to transformed
+ *  to normal fixed dimenson tensor
+ * \author Tianqi Chen
+ */
+#ifndef MXNET_TENSOR_BLOB_H_
+#define MXNET_TENSOR_BLOB_H_
+
+#include <vector>
+#include <iostream>
+#include <utility>
+#include <algorithm>
+#include <dmlc/logging.h>
+#include "./base.h"
+
+namespace mxnet {
```

Comment:
This is likely to be fine, as originally we typedefed TShape inside mxnet namespace